crates/turborepo-lib/src/diagnostics.rs

```diff
@@ -0,0 +1,523 @@
+    async fn log_line(chan: &MpscSender<DiagnosticMessage>, message: String) {
+        chan.send(DiagnosticMessage::LogLine(
+            Self::NAME.to_string(),
+            message.to_string(),
+        ))
+        .await
+        .ok(); // channel closed, ignore
+    }
+
+    /// prompts the user with a message and options
+    async fn request(
+        chan: &MpscSender<DiagnosticMessage>,
+        message: String,
+        options: Vec<String>,
+    ) -> Option<Receiver<String>> {
+        let (tx, rx) = oneshot::channel();
+        chan.send(DiagnosticMessage::Request(
+            Self::NAME.to_string(),
+            message.to_string(),
+            options,
+            tx,
+        ))
+        .await
+        .ok()?; // if the channel is closed, we can't request
+
+        Some(rx)
+    }
+
+    /// suspends the terminal output and returns a pair of channels
+    ///
+    /// the first one signifies to the diagnostic when it is safe to render
+    /// and the second one is used by the diagnostic to notify the renderer
+    /// when it is done
+    async fn suspend(chan: &MpscSender<DiagnosticMessage>) -> Option<(Receiver<()>, Sender<()>)> {
+        let (stopped_tx, stopped_rx) = oneshot::channel();
+        let (resume_tx, resume_rx) = oneshot::channel();
+
+        chan.send(DiagnosticMessage::Suspend(
+            Self::NAME.to_string(),
+            stopped_tx,
+            resume_rx,
+        ))
+        .await
+        .ok()?; // if the channel is closed, we can't suspend
+
+        Some((stopped_rx, resume_tx))
+    }
+
+    // types of exit
+
+    /// the diagnostic is done
+    async fn done(chan: MpscSender<DiagnosticMessage>, message: String) {
+        chan.send(DiagnosticMessage::Done(
+            Self::NAME.to_string(),
+            message.to_string(),
+        ))
+        .await
+        .ok(); // channel closed, ignore
+    }
+
+    /// the diagnostic failed
+    async fn failed(chan: MpscSender<DiagnosticMessage>, message: String) {
+        chan.send(DiagnosticMessage::Failed(
+            Self::NAME.to_string(),
+            message.to_string(),
+        ))
+        .await
+        .ok(); // channel closed, ignore
+    }
+
+    /// the diagnostic is not applicable
+    async fn not_applicable(chan: MpscSender<DiagnosticMessage>, message: String) {
+        chan.send(DiagnosticMessage::NotApplicable(
+            Self::NAME.to_string(),
+            message.to_string(),
+        ))
+        .await
+        .ok(); // channel closed, ignore
+    }
+}
+
+pub struct GitDaemonDiagnostic;
+
+impl Diagnostic for GitDaemonDiagnostic {
+    const NAME: &'static str = "git.daemon";
+
+    fn execute(&self, chan: MpscSender<DiagnosticMessage>) {
+        tokio::task::spawn(async move {
+            if !(cfg!(target_os = "windows") || cfg!(target_os = "macos")) {
+                // the git daemon is not implemented on unix
+                Self::not_applicable(chan, "Git FS Monitor (not available on unix)".to_string())
```

Comment:
Is it Unix or Linux? macOS is Unix based, no?

Reply:
Yes, oops